docs/docs/dev_guide/index.md

```diff
@@ -0,0 +1,116 @@
+
+Gitpod provides the ability to initialize the environment with shell commands and scripts. We leverage these init commands to prime a development environment (terminal) for the AaC Python package and the AaC VSCode Extension (Typescript). You can find the init commands in the `.Gitpod.yml` file at the root of the repository.
+
+## Non-Gitpod Environment
+
+### Requirements
+There are some developmental and system requirements for developing with AaC. The always up-to-date source of truth for requirements will be the `.gitpod.Dockerfile` at the root of the AaC repository.
+* Python 3.9+ Installed
+* Typescript Dependencies
+    * libnss3-dev
+    * libgbm-dev
+
+### Developing the AaC Python Package
+The AaC python package, its source code, and its configuration are all found in the `python/` directory.
+
+#### Installing AaC Python Package
+We expect python developers to use a virtual environment when developing the AaC package, and so the project is configured only for use with virtual environments.
+
+Create a new environment:
+```bash
+python3.9 -m venv venv --system-site-packages
+```
+
+Configure your shell with the virtual environment:
+```bash
+source venv/bin/activate
+```
+
+Install the AaC package in development mode:
+```bash
+pip install -e .[all]
+```
+
+Run the AaC `version` command to confirm the package is installed:
+```bash
+aac version
+```
+
+<br>
+
+#### AaC Python Package Dependency Control
+For more fine-grained control when installing dependencies, they have been structured into 3 sections:
+- runtime - the dependencies required to run the project
+- test - the dependencies required to run the project's automated tests
+- dev - the dependencies required to run development tools like linting and quality checks
+
+To install only runtime dependencies simply use pip to install the dependencies:
+```bash
+$ pip install -e .
+```
+
+If you'd like to install the additional dependencies (test and dev), then you can specify that pip include those dependencies like such:
+
+To install test dependencies (test), run this:
+```bash
+$ pip install -e .[test]
+```
+
+To install development dependencies (dev), run this:
+```bash
+$ pip install -e .[dev]
+```
+
+To install all dependencies (runtime, dev, and test), run this:
+```bash
+$ pip install -e .[all]
+```
+
+#### Testing the Python Package
+To run tests, make sure you've set up your dependencies using `pip install -e .[test]` or `pip install -e .[all]`. Then, from the root of the python project, run the following shell command (from within your virtual environment).
+
+```bash
+$ tox
+```
+
+#### VSCode Unit Test Support for Python
+The python plugin for VSCode supports the `unittest` python library, and not the nose2 framework that we use. The impact is fairly minimal, but python unit tests using the nose2 parameterized tests will not run correctly from within the VSCode IDE since it fails to recognize the nose2 parameterized tests as parameterized tests causing them to fail.
```

Comment:
```suggestion
The python plugin for VSCode supports the `unittest` python library, and not the `nose2` framework that we use. The impact is fairly minimal, but python unit tests using the `nose2` parameterized test feature will not run correctly from within the VSCode IDE since it fails to recognize the `nose2` parameterized tests as parameterized tests, they will fail.

To get around this, you can run all the tests from the terminal using the above `tox` commands.
```

Reply:
Tiny edit. How about this?
```suggestion
The python plugin for VSCode supports the `unittest` python library, and not the `nose2` framework that we use. The impact is fairly minimal, but python unit tests using the `nose2` parameterized test feature will not run correctly from within the VSCode IDE since it fails to recognize the `nose2` parameterized tests as parameterized tests causing them to fail.
	
To get around this, you can run all the tests from the terminal using the above `tox` commands.
```